### ISSUE TITLE
Add Cobaya Theory interface

### ISSUE THREAD

**Issue by gcanasherrera** [cloe-org/cloelib#38]
## 📝 Issue Summary

cloelib is a self-standing theory code that could be potentially be wrapped around Cobaya as a theory code using Cobaya's [theory class](https://cobaya.readthedocs.io/en/latest/theory.html).

The simple way to do this would be to create a script/notebook where we play how to wrap cloelib around Cobaya's theory class so that we can use the model wrapper of Cobaya itself to produce photometric Cells predictions, for instance.

### ✅ Expected Outcome

- **Goal 1**: a script example of how to wrap this up around Cobaya
- **Goal 2**: start discussing with Cobaya maintainers how we could like this integrated into Cobaya 

### 📌 Additional Notes
Pinging here some amazing people that have been playing with Cobaya recently: @LisaGoh @Alkistis @chiaradeleo1 @matmartinelli

**Comment by gcanasherrera** [comment]
Could we start discussing here how to do this? @LisaGoh @Alkistis @chiaradeleo1 @matmartinelli

**Comment by matmartinelli** [comment]
In @chiaradeleo1 [CANDI code](https://github.com/chiaradeleo1/CANDI) we relied on a module that was wrapping up our theory code to Cobaya.

It was quite simple in our case, but it could be a place to start. You can find the wrapper here: https://github.com/chiaradeleo1/CANDI/blob/main/theory_code/cobaya_theory_wrapper.py

**Comment by gcanasherrera** [comment]
Exactly, thanks @matmartinelli! This is what I had in mind! I think need to think a bit further into how to get the protocols working with CAMB/CLASS way of Cobaya of retrieving things.
